### DISCUSSION TITLE
Zustand fetching data inside store

### DISCUSSION THREAD

**Discussion by Crim-oo** [pmndrs/zustand#1154] [Q&A] [answered]
### store/auth.ts
      export type UserStatus = "idle" | "signOut" | "signIn";
      
      export interface AuthState {
        currentUser: User | undefined;
        token: string | undefined;
        status: UserStatus;
        signUpWithCreds: (email: string, password: string) => void;
      }
      
      const useAuthStoreBase = create<AuthState>()((set) => ({
        currentUser: undefined,
        token: undefined,
        status: "idle",
        signUpWithCreds: async (email: string, password: string) => {
          createUserWithEmailAndPassword(auth, email, password)
            .then(async (userCred: UserCredential) => {
              const user = userCred.user;
              const token = await user.getIdToken();
              set((state) => ({ currentUser: user, token: token }));
              // Implement the rest of the logic
            })
            .catch((err) => {
              // Implement the rest of the logic
            });
        },
      }));

So i'm new to the dev world and i wanted to know if fetching data from a server inside the store is considered as a bad practice.

**Comment by dai-shi** [comment, marked as answer]
It's good: https://github.com/pmndrs/zustand#async-actions
